### ISSUE TITLE
Avatar thumbnails are shuffled

### ISSUE THREAD

**Issue by Gakuto1112** [FiguraMC/Figura#198]
## Description
Avatars in the avatar list shows their thumbnail images, but they will show other avatar's thumbnail after performing certain operations.

![The screenshot that describes this issue](https://github.com/FiguraMC/Figura/assets/90630001/afa7dd20-866e-446d-ab08-dcac1d0d3b79)

In above screenshot, correct thumbnails are followings:

| Avatar name | Correct thumbnail |
| - | - |
| シロ (Shiro) | <img src="https://raw.githubusercontent.com/Gakuto1112/SenkoSan/Shiro/avatar.png" width="64"> |
| 仙狐さん (Senko-san) | <img src="https://raw.githubusercontent.com/Gakuto1112/SenkoSan/Senko/avatar.png" width="64"> |
| 夜空 (Sora) | <img src="https://raw.githubusercontent.com/Gakuto1112/SenkoSan/Sora/avatar.png" width="64"> |
| 鈴 (Suzu) | <img src="https://raw.githubusercontent.com/Gakuto1112/SenkoSan/Suzu/avatar.png" width="64"> |

Thumbnails that avatars shows incorrectly are of other avatars.

## How to reproduce
1. Create many avatars,
    - I tested with [this set of avatars](https://github.com/FiguraMC/Figura/files/14504112/avatars.zip).
3. Save those avatars with subfolders included.
4. Go to the avatar wardrobe.
5. Open subfolders.
6. Scroll the avatar list down and up.
7. You will see that some thumbnails are wrong. 

## Expected behaviors
Avatars show their thumbnails correctly even if the above operations are performed.

## Environments
| Item | Version |
| - | - |
| Minecraft | 1.20.4 |
| Fabric Loader | 0.15.7 |
| Fabric API | 0.96.4+1.20.4 |
| Figura | 0.1.4+1.20.4 |

This issue is also occurred in older Figura versions.

**Comment by PoolloverNathan** [comment]
Can confirm — I have several avatars with the same icon (a pack) and nearby avatars also have that icon. I wonder how subfolders have an effect… maybe I should try looking into ~~the depths of hell~~ Figura's code again at some point.

**Comment by UnlikePaladin** [comment]
I looked briefly into it and I think the code for getting the icon is just bad, it probably needs to be rewritten

**Comment by ToastTextures** [comment]
Is getting the icon from a zipped avatar bad™ because of that code

**Comment by UnlikePaladin** [comment]
yes, I'm like 99% sure the zips are the problem, the code doesn't account for it well and I think it results in undefined behavior
